shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -275,12 +296,79 @@ bool FlutterWindowsEngine::RunWithEntrypoint(const char* entrypoint) {
               << std::endl;
     return false;
   }
+  FlutterEngineDisplay display;
+  display.struct_size = sizeof(FlutterEngineDisplay);
+  display.display_id = 0;
+  display.single_display = true;
+  display.refresh_rate = ApproxFrameRate();
+
+  std::vector<FlutterEngineDisplay> displays = {display};
+  embedder_api_.NotifyDisplayUpdate(engine_,
+                                    kFlutterEngineDisplaysUpdateTypeStartup,
+                                    displays.data(), displays.size());
 
   SendSystemSettings();
 
   return true;
 }
 
+uint64_t SnapToNextTick(uint64_t value,
+                        uint64_t tick_phase,
+                        uint64_t tick_interval) {
+  uint64_t offset = (tick_phase - value) % tick_interval;
+  if (offset != 0)
+    offset = offset + tick_interval;
+  return value + offset;
+}
+
+void FlutterWindowsEngine::OnVsync(intptr_t baton) {
+  uint64_t current_time = embedder_api_.GetCurrentTime();
+  int64_t interval;
+  uint64_t next_time;
+  DWM_TIMING_INFO timing_info;
+
+  timing_info.cbSize = sizeof(timing_info);
+  HRESULT result = DwmGetCompositionTimingInfo(NULL, &timing_info);
+  if (result == S_OK && qpc_supported_) {
+    LARGE_INTEGER current_ticks;
+    QueryPerformanceCounter(&current_ticks);
+    double cycle_delta =
+        static_cast<double>((timing_info.qpcVBlank - current_ticks.QuadPart) *
+                            1000000000) /
+        lp_frequency_.QuadPart;
+    double local_interval =
+        static_cast<double>(timing_info.qpcRefreshPeriod * 1000000000) /
+        lp_frequency_.QuadPart;
+    interval = llround(local_interval);
+    next_time = current_time + cycle_delta;
+    if (next_time > current_time + interval) {
+      next_time = current_time + interval;
+    }
```

Comment:
This is all kind of hacky for now. chrome has a better approach that I'll port eventually, but I don't think this is the cause of missed frames